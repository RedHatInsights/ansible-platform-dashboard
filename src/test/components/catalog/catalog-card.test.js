import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { MemoryRouter } from 'react-router-dom';
import notificationsMiddleware from '@redhat-cloud-services/frontend-components-notifications/notificationsMiddleware';
import catalogReducer, { catalogInitialState } from '../../../redux/reducers/catalog-reducer';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { applyReducerHash, ReducerRegistry } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
import CatalogCard from '../../../components/catalog/catalog-card';
import UserContext from '../../../user-context';
import '../../../helpers/shared/__mocks__/user-login';

const ComponentWrapper = ({ store, initialEntries = [ '/ansible-dashboard' ], children }) => (
  <IntlProvider locale="en">
    <Provider store={ store } >
      <MemoryRouter initialEntries={ initialEntries }>
        <UserContext.Provider
          value={ { permissions: []} }
        >
          { children }
        </UserContext.Provider>
      </MemoryRouter>
    </Provider>
  </IntlProvider>
);

describe('<AnalyticsCard />', () => {
  let initialProps;
  let initialState;
  const middlewares = [thunk, promiseMiddleware, notificationsMiddleware()];
  let mockStore;

  beforeEach(() => {
    global.insights = {
      chrome: {
        identifyApp: jest.fn(),
        init: jest.fn(),
        auth: {
          getUser: () => Promise.resolve(true)
        }
      }
    };
    initialProps = {};
    mockStore = configureStore(middlewares);
    initialState = {
      i18nReducer: {
        formatMessage: ({ defaultMessage }) => defaultMessage
      },
      analyticsReducer: { ...catalogInitialState, isLoading: false },
      catalogReducer: { ...catalogInitialState, isLoading: false }
    };
  });

  it('should render correctly', async ()=> {
    const store = mockStore(initialState);
    const registry = new ReducerRegistry({}, [ thunk, promiseMiddleware ]);
    registry.register({ analyticsReducer: applyReducerHash(catalogReducer, catalogInitialState) });

    const { asFragment } = render(<ComponentWrapper store={ store }><CatalogCard { ...initialProps } /></ComponentWrapper>);

    await waitFor(() => expect(() => screen.getByLabelText('Contents')).toThrow());

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly in loading state', async () => {
    const store = mockStore(initialState);

    const { asFragment } = render(<ComponentWrapper store={ store }><CatalogCard { ...initialProps } isLoading={ true } /></ComponentWrapper>);

    expect(asFragment()).toMatchSnapshot();

    await waitFor(() => expect(() => screen.getByLabelText('Contents')).toThrow());
  });
});
