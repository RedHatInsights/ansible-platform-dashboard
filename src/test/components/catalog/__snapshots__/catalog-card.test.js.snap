// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnalyticsCard /> should render correctly 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card ans-c-card-dashboard"
    data-ouia-component-id="OUIA-Generated-Card-1"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-c-card__title"
    >
      <h3
        class="pf-c-title pf-m-lg"
        data-ouia-component-id="OUIA-Generated-Title-1"
        data-ouia-component-type="PF4/Title"
        data-ouia-safe="true"
      >
        Automation Services Catalog
      </h3>
    </div>
    <div
      class="pf-c-card__body"
    >
      <div
        class="pf-l-flex ans-l-flex ans-l-flex-automation-hub-card"
      >
        <div
          class="pf-l-flex"
        >
          <div
            class="pf-c-content"
          >
            <p
              class=""
              data-ouia-component-id="OUIA-Generated-Text-1"
              data-ouia-component-type="PF4/Text"
              data-ouia-safe="true"
              data-pf-content="true"
            >
              Collect and distribute automation content, govern content by approval processes and assure sign-off by assigned groups. 
              <br />
              <br />
            </p>
          </div>
          <div
            class="pf-l-flex"
          >
            <div
              class="pf-l-flex pf-m-column pf-u-m-0 pf-u-pr-sm"
            >
              <div
                class="pf-m-align-right pf-u-mb-sm pf-u-mt-md"
              >
                <div
                  class="pf-c-content"
                >
                  <h1
                    class=""
                    data-ouia-component-id="OUIA-Generated-Text-2"
                    data-ouia-component-type="PF4/Text"
                    data-ouia-safe="true"
                    data-pf-content="true"
                  >
                    0
                  </h1>
                </div>
              </div>
              <div
                class="pf-m-align-right pf-u-mb-sm pf-u-mt-md"
              >
                <div
                  class="pf-c-content"
                >
                  <h1
                    class=""
                    data-ouia-component-id="OUIA-Generated-Text-3"
                    data-ouia-component-type="PF4/Text"
                    data-ouia-safe="true"
                    data-pf-content="true"
                  >
                    0
                  </h1>
                </div>
              </div>
            </div>
            <div
              class="pf-l-flex pf-m-column"
            >
              <div
                class=""
              >
                <a
                  aria-disabled="false"
                  class="pf-c-button pf-m-link"
                  data-ouia-component-id="OUIA-Generated-Button-link-1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  href="/ansible/catalog/products"
                >
                  Products
                </a>
              </div>
              <div
                class=""
              >
                <a
                  aria-disabled="false"
                  class="pf-c-button pf-m-link"
                  data-ouia-component-id="OUIA-Generated-Button-link-2"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  href="/ansible/catalog/portfolios"
                >
                  Portfolios
                </a>
              </div>
            </div>
          </div>
        </div>
        <hr
          class="pf-c-divider"
        />
        <div
          class="pf-l-flex"
        >
          <div
            class="pf-l-flex pf-m-column pf-m-align-self-stretch"
          >
            <h4
              class="pf-c-title pf-m-md"
              data-ouia-component-id="OUIA-Generated-Title-2"
              data-ouia-component-type="PF4/Title"
              data-ouia-safe="true"
            >
              Latest created product
            </h4>
            <br />
          </div>
        </div>
        <hr
          class="pf-c-divider"
        />
        <div
          class="pf-l-flex pf-m-column"
        >
          <div
            class=""
          >
            <div
              class="pf-l-flex"
            >
              <div
                class=""
              >
                <h4
                  class="pf-c-title pf-m-md"
                  data-ouia-component-id="OUIA-Generated-Title-3"
                  data-ouia-component-type="PF4/Title"
                  data-ouia-safe="true"
                >
                  Latest orders
                </h4>
              </div>
            </div>
          </div>
          <div
            class="pf-l-flex pf-m-column"
          >
            <div
              class="pf-m-align-self-center"
            >
              <div
                class="pf-c-content"
              >
                <h6
                  class=""
                  data-ouia-component-id="OUIA-Generated-Text-4"
                  data-ouia-component-type="PF4/Text"
                  data-ouia-safe="true"
                  data-pf-content="true"
                >
                  No orders yet
                </h6>
              </div>
            </div>
            <div
              class="pf-m-align-self-center"
            >
              <div
                class="pf-c-content"
              >
                <small
                  class=""
                  data-ouia-component-id="OUIA-Generated-Text-5"
                  data-ouia-component-type="PF4/Text"
                  data-ouia-safe="true"
                  data-pf-content="true"
                >
                  Orders show up here when an order is placed.
                </small>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </article>
</DocumentFragment>
`;

exports[`<AnalyticsCard /> should render correctly in loading state 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card ans-c-card-dashboard"
    data-ouia-component-id="OUIA-Generated-Card-2"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-c-card__title"
    >
      <h3
        class="pf-c-title pf-m-lg"
        data-ouia-component-id="OUIA-Generated-Title-4"
        data-ouia-component-type="PF4/Title"
        data-ouia-safe="true"
      >
        Automation Services Catalog
      </h3>
    </div>
    <div
      class="pf-c-card__body"
    >
      <section
        class=""
        style="background-color: white; min-height: 100%;"
      >
         
        <div
          class="pf-l-bullseye"
        >
          <svg
            aria-label="Contents"
            aria-valuetext="Loading..."
            class="pf-c-spinner pf-m-xl"
            role="progressbar"
            viewBox="0 0 100 100"
          >
            <circle
              class="pf-c-spinner__path"
              cx="50"
              cy="50"
              fill="none"
              r="45"
            />
          </svg>
        </div>
         
      </section>
    </div>
  </article>
</DocumentFragment>
`;
